### PR TITLE
[jax2tf] Added a non-XLA conversion path for reduce_window_sum

### PR DESCRIPTION
**Pull request by qiuminxu**
PR#4589 added a conversion path for reduce_window_max use tf.nn.max_pool.
This PR make the function broader to also apply to reduce_window_sum (which is used by flax.nn.avg_pool).

### CODE REVIEW

File: jax/experimental/jax2tf/jax2tf.py [hunk 1]
```diff
@@ -1596,24 +1596,30 @@ def reducer(arg1: TfVal, arg2: TfVal) -> TfVal:
...
   # Contrarily to the main path, tf.int8 is actually a valid type for
   # tf.nn.max_pool.
-  if operand.dtype in [tf.bool, tf.uint32, tf.uint64, tf.complex64,
-                       tf.complex128]:
+  if op_name == "reduce_window_max" and operand.dtype in [
+      tf.bool, tf.uint32, tf.uint64, tf.complex64, tf.complex128
+  ]:
     return f"tf.nn.max_pool does not support operands of type {operand.dtype}"
+  if op_name == "reduce_window_sum" and operand.dtype not in [
+      tf.half, tf.float16, tf.float32, tf.float64
```
Comment (bchetioui) [inline jax/experimental/jax2tf/jax2tf.py:1615]: I don't think there is a difference between `tf.half` and `tf.float16`:
```
>>> tf.half is tf.float16
True
```
I think we could just use `tf.float16`? :)

**Review by bchetioui — APPROVED** [review]
Passes the tests otherwise, LGTM once the `tf.half`/`tf.float16` thing is fixed :)

File: jax/experimental/jax2tf/jax2tf.py (hunk 1 above)
Comment (gnecula) [reply jax/experimental/jax2tf/jax2tf.py:1615]: I would also prefer tf.float16; I do not think we have any uses of tf.half in the code base.

**Review by gnecula — APPROVED** [review]

File: jax/experimental/jax2tf/jax2tf.py (hunk 1 above)
Comment (qiuminxu) [reply jax/experimental/jax2tf/jax2tf.py]: Yep, I missed this. Done.

**Pull request merged**